pkg/lockfile/parse-requirements-txt.go

```diff
@@ -105,10 +109,16 @@ func ParseRequirementsTxt(pathToLockfile string) ([]PackageDetails, error) {
 	for scanner.Scan() {
 		line := removeComments(scanner.Text())
 
-		if strings.HasPrefix(line, "-r ") {
-			details, err := ParseRequirementsTxt(
-				filepath.Join(filepath.Dir(pathToLockfile), strings.TrimPrefix(line, "-r ")),
-			)
+		if ar := strings.TrimPrefix(line, "-r "); ar != line {
+			ar = filepath.Join(filepath.Dir(pathToLockfile), ar)
```

Comment:
is there some path expansion/resolution that that would deal with e.g. `./../pip/cyclic-r-complex-1.txt` is the same as `./../pip/../pip/cyclic-r-complex-1.txt`

Reply:
Yup that's handled by `filepath.Join` which [Cleans the path after the join](https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/path/filepath/path_unix.go;l=49)